ime/app/src/main/java/com/anysoftkeyboard/keyboards/views/PointerTracker.java

```diff
@@ -466,15 +466,41 @@ private void detectAndSendKey(int index, int x, int y, long eventTime) {
                 listener.onCancel();
             }
         } else {
-            if (key.text != null && !mKeyDetector.isKeyShifted(key)) {
+            if (key.text != null || (key.shiftedText != null && mKeyDetector.isKeyShifted(key))) {
                 if (listener != null) {
-                    listener.onText(key, key.text);
-                    listener.onRelease(0); // dummy key code
-                }
-            } else if (key.shiftedText != null && mKeyDetector.isKeyShifted(key)) {
-                if (listener != null) {
-                    listener.onText(key, key.shiftedText);
-                    listener.onRelease(0); // dummy key code
+                    CharSequence text = null;
+                    if (!mKeyDetector.isKeyShifted(key)) {
+                        text = key.text;
+                    } else if (mKeyDetector.isKeyShifted(key)) {
+                        text = key.shiftedText;
+                    }
+                    int[] nearByKeyCodes = mKeyDetector.newCodeArray();
```

Comment:
I was wondering about the near-by-codes when you created the issue.
Maybe we do not need that at all? Maybe we can provide a single (new) array here with just the code? These near-by-codes are used mostly to generate corrections due to user pressing a neighbor key instead of the key they actually wanted.

Reply:
Would we get rid of that entirely, then?

Reply:
yes, I think so..